### PR TITLE
refactor: Modified from_json in AmbiguityConfigJson for easier implementation in Athena

### PR DESCRIPTION
**Pull request by Ragansu**
This PR is to do 2 changes for easier implementation in Athena
a) Change factor_hits/holes to goodhits/holes and fakehits/holes this is how config is defined in Athena
b) Change ERROR to DEBUG in track state with no surface as Athena has extra trackstates which doesnt have surface. 

Ragansu

### CODE REVIEW

File: Core/src/AmbiguityResolution/ScoreBasedAmbiguityResolution.cpp [hunk 1]
```diff
@@ -19,7 +19,7 @@ std::vector<bool> Acts::ScoreBasedAmbiguityResolution::getCleanedOutTracks(
     const {
   std::vector<bool> cleanTracks(measurementsPerTrack.size(), false);
 
-  ACTS_VERBOSE("Cleaning tracks");
+  ACTS_INFO("Cleaning tracks");
```
Comment (CarloVarni) [inline Core/src/AmbiguityResolution/ScoreBasedAmbiguityResolution.cpp:22]: I'd keep `VERBOSE`

File: Plugins/Json/src/AmbiguityConfigJsonConverter.cpp [hunk 2]
```diff
@@ -35,15 +35,26 @@ void from_json(const nlohmann::json& j, ConfigPair& p) {
 
     detectorConfig.sharedHitsFlag = value["sharedHitsFlag"];
 
-    std::vector<double> factorHits = value["factorHits"];
-    std::vector<double> factorHoles = value["factorHoles"];
+    std::vector<double> goodHits = value["goodHits"];
+    std::vector<double> goodHoles = value["goodHoles"];
```
Comment (CarloVarni) [inline Plugins/Json/src/AmbiguityConfigJsonConverter.cpp:39]: are we ok with copies?

File: Plugins/Json/src/AmbiguityConfigJsonConverter.cpp [hunk 3]
```diff
@@ -35,15 +35,26 @@ void from_json(const nlohmann::json& j, ConfigPair& p) {
...
 
-    for (auto factor : factorHits) {
-      detectorConfig.factorHits.push_back(factor);
+    std::vector<double> fakeHits = value["fakeHits"];
+    std::vector<double> fakeHoles = value["fakeHoles"];
```
Comment (CarloVarni) [inline Plugins/Json/src/AmbiguityConfigJsonConverter.cpp:42]: why assign them every single time? allegedly they are always the same numbers, right? Also, why a copy?

Comment (Ragansu) [reply Plugins/Json/src/AmbiguityConfigJsonConverter.cpp:42]: So you think its better to have a fixed config in a header file in athena than use the from json file to load it every time ? 
The core algorithm doesn't care how the detector config is assigned. So from_json is in the sense an optional tool.

Comment (CarloVarni) [reply Plugins/Json/src/AmbiguityConfigJsonConverter.cpp:42]: my point was that this assignment is done inside a loop. But I assume that `value["fakeHits"]` remains the same for every iteration. Right?

File: Plugins/Json/src/AmbiguityConfigJsonConverter.cpp (hunk 2 above)
Comment (CarloVarni) [reply Plugins/Json/src/AmbiguityConfigJsonConverter.cpp:39]: what about having these as `const` as well?

```suggestion
    const std::vector<double>& goodHits = value["goodHits"];
    const std::vector<double>& goodHoles = value["goodHoles"];
```

File: Plugins/Json/src/AmbiguityConfigJsonConverter.cpp (hunk 3 above)
Comment (CarloVarni) [reply Plugins/Json/src/AmbiguityConfigJsonConverter.cpp:42]: ah no sorry... I misunderstood the code. Forget about this point

Comment (CarloVarni) [inline Plugins/Json/src/AmbiguityConfigJsonConverter.cpp:42]: the question remains if these can be const references or if we really need copies.
Can this be made a 
```suggestion
    const std::vector<double>& fakeHits = value["fakeHits"];
    const std::vector<double>&fakeHoles = value["fakeHoles"];
```

**Review by CarloVarni — APPROVED** [review]

**Comment by acts-project-service** [comment]
## :white_check_mark: Athena integration test results [971be3af6f4b36fed048dc129031fde1c79bfabc]
### :white_check_mark: All tests successful

|status|job|report|
|------|---|------|
|:green_circle: | [report_pull_request](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/43744968)||
|:green_circle: | [run_art_test: test_data18_13TeV_1000evt](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/43744967)||
|:green_circle: | [run_art_test: test_ttbarPU40_reco](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/43744966)||
|:green_circle: | [test_ActsDumpGeometryIdentifiers](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/43744965)||
|:green_circle: | [test_ActsCheckObjectCountsCachedWorkflow](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/43744964)||
|:green_circle: | [test_ActsCheckObjectCountsWorkflow](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/43744963)||
|:green_circle: | [test_ActsEFTrackFit](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/43744962)||
|:green_circle: | [test_ActsPersistifySeeds](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/43744961)||
|:green_circle: | [test_ActsBenchmarkWithSpot](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/43744960)||
|:green_circle: | [test_ActsAnalogueClustering](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/43744959)||
|:green_circle: | [test_ActsWorkflowHeavyIons](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/43744958)||
|:green_circle: | [test_ActsWorkflowFastTracking](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/43744957)||
|:green_circle: | [test_ActsWorkflowCached](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/43744956)||
|:green_circle: | [test_ActsWorkflow](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/43744955)||
|:green_circle: | [test_ActsValidateAmbiguityResolution](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/43744954)||
|:green_circle: | [test_ActsValidateResolvedTracks](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/43744953)||
|:green_circle: | [test_ActsValidateTracks](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/43744952)||
|:green_circle: | [test_ActsValidateActsCoreSpacePoints](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/43744951)||
|:green_circle: | [test_ActsValidateActsSpacePoints](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/43744950)||
|:green_circle: | [test_ActsValidateSeeds](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/43744949)||
|:green_circle: | [test_ActsValidateOrthogonalSeeds](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/43744948)||
|:green_circle: | [test_ActsValidateClusters](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/43744947)||
|:green_circle: | [test_ActsPersistifyEDM](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/43744946)||
|:green_circle: | [test_ActsGSFRefitting](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/43744945)||
|:green_circle: | [test_ActsKfRefitting](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/43744944)||
|:green_circle: | [test_ActsExtrapolationAlgTest](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/43744943)||
|:green_circle: | [test_ActsITkTest](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/43744942)||
|:green_circle: | [run_workflow_tests_run4_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/43744941)||
|:green_circle: | [run_workflow_tests_run2_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/43744940)||
|:green_circle: | [run_workflow_tests_run2_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/43744939)||
|:green_circle: | [run_workflow_tests_run3_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/43744938)||
|:green_circle: | [run_workflow_tests_run3_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/43744937)||
|:green_circle: | [run_unit_tests](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/43744936)||

**Pull request merged**